xmodule/item_bank_block.py

```diff
@@ -0,0 +1,503 @@
+"""
+Definitions for ItemBankMixin and ItemBankBlock.
+"""
+from __future__ import annotations
+
+import json
+import logging
+import random
+from copy import copy
+from django.conf import settings
+from django.utils.functional import classproperty
+from lxml import etree
+from lxml.etree import XMLSyntaxError
+from rest_framework import status
+from web_fragments.fragment import Fragment
+from webob import Response
+from xblock.completable import XBlockCompletionMode
+from xblock.core import XBlock
+from xblock.fields import Boolean, Integer, List, Scope, String
+
+from xmodule.mako_block import MakoTemplateBlockBase
+from xmodule.studio_editable import StudioEditableBlock
+from xmodule.util.builtin_assets import add_webpack_js_to_fragment
+from xmodule.validation import StudioValidation, StudioValidationMessage
+from xmodule.xml_block import XmlMixin
+from xmodule.x_module import (
+    ResourceTemplates,
+    XModuleMixin,
+    XModuleToXBlockMixin,
+    shim_xmodule_js,
+    STUDENT_VIEW,
+)
+
+_ = lambda text: text
+
+logger = logging.getLogger(__name__)
+
+
+@XBlock.needs('mako')
+@XBlock.wants('user')
+class ItemBankMixin(
+    # TODO: Whittle down list of mixins if possible.
+    # https://github.com/openedx/edx-platform/issues/35686
+    MakoTemplateBlockBase,
+    XmlMixin,
```

Comment:
It seems you can remove `XmlMixin` now. `ItemBankBlock` works fine without it AFAICT, although you may have to update some test cases that rely on its weird default behavior of writing the XML to a file in `export_fs`.

(edit: it may also affect import/export in some way, but I'm not sure)

Reply:
Yeah, I'll do this in a follow-up